test/test.js

```diff
@@ -244,76 +687,216 @@ test('ogrChild module', function(t){
+  t.ok(isStream(badPipeline), 'Bad pipeline still returns stream');
+
+  function trunc(n, digits){
+    var shift = Math.pow(10, digits);
+    var func = n < 0 ? Math.ceil : Math.floor;
+    return func(n * shift)/shift;
+  }
+
+  function checkPipes(pipeline, stats){
+    pump(pipeline, stats, stats.sink(), function(err){
+      t.notOk(err, 'Doesn\'t err in a correct pipeline');
+
+      var store = JSON.parse(stats.stats.store.toString());
+      store.geometry.coordinates[0] = trunc(store.geometry.coordinates[0], 5)
+      store.geometry.coordinates[1] = trunc(store.geometry.coordinates[1], 5)
+      store = JSON.stringify(store);
+
+      t.equal(store, ncjson, 'Transforms properly in retriever pipeline when file is streamed');
+    });
+  }
+
+  checkPipes(pipeline, pipeStats);
+  checkPipes(pipe2, pipeStats2);
+
+  badPipeline.once('error', function(err){
+    t.ok(err, 'Passes through error when encountered in the pipeline');
+  });
+
+});
+
+
+
+
+test('loader-pipeline module', function(t){
+  t.plan(5);
+
+  var record = fs.readJsonSync('test/data/metadata/ncmeta.json');
+
+  var client = options.client;
+  var op1 = options;
+  op1.client = null;
+  var op2 = JSON.parse(JSON.stringify(options));
+  op1.client = client;
+  op2.client = client;
+  op2.forcedIndex = 'testforcedindex';
+
+  var str1 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+  var str2 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+  var str3 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+
+  loaderPipeline(op1, str1, record, function(err, loader){
+    t.notOk(err, 'No error on valid loader pipeline with derived index.');
+    t.ok(loader, 'Returns loader stream');
+    loader.on('error', function(err){t.fail(err)});
+  });
+
+  loaderPipeline(op2, str2, record, function(err, loader){
+    t.notOk(err, 'No error on valid loader pipeline with forced index.');
+    t.ok(loader, 'Returns loader stream');
+    loader.on('error', function(err){t.fail(err)});
+  });
+
+  loaderPipeline({}, str3, record, function(err){
+    t.ok(err, 'Error propagated on bad options.');
+  });
+
+});
+
+
+
+
+test('loader', function(t){
+  t.plan(4);
+
+  var record = fs.readJsonSync('test/data/metadata/ncmeta.json');
+
+  var client = options.client;
+  var op1 = options;
+  op1.client = null;
+  var op2 = JSON.parse(JSON.stringify(options));
+  op1.client = client;
+  op2.client = client;
+  op2.logger = logger;
+  op2.forcedIndex = 'testforcedindex';
+
+  var str1 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+  var str2 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+  var str3 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+  var str4 = retrieverPipeline(record, 'test/data/fields/north_carolina.json');
+
+  loader(op1, str1, record, function(err){
+    t.notOk(err, 'Loads without error');
```

Comment:
Should these test messages be more specific?  How do you know one from the next?